services/f-droid/f-droid.service.js

```diff
@@ -1,31 +1,92 @@
 'use strict'
 
+const Joi = require('joi')
+const yaml = require('js-yaml')
 const BaseService = require('../base')
+const validate = require('../../lib/validate')
 const { addv: versionText } = require('../../lib/text-formatters')
 const { version: versionColor } = require('../../lib/color-formatters')
-const { InvalidResponse } = require('../errors')
+const { InvalidResponse, InvalidParameter } = require('../errors')
 
 module.exports = class FDroid extends BaseService {
-  async fetch({ appId }) {
-    // currently, we only use the txt format. There are few apps using the yml format.
-    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}.txt`
-    const { buffer } = await this._request({
-      url,
+  static render({ version }) {
+    return {
+      message: versionText(version),
+      color: versionColor(version),
+    }
+  }
+
+  async handle({ appId }, queryParams) {
+    const constructor = this.constructor
+    const { metadata_format: format } = constructor.validateParams(queryParams)
+    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}`
+    const fetchOpts = {
       options: {},
       errorMessages: {
         404: 'app not found',
       },
+    }
+    const fetch = format === 'yml' ? this.fetchYaml : this.fetchText
+    let result
+
+    try {
+      // currently, we only use the txt format to the initial fetch because
+      // there are more apps with that format but yml is now the standard format
+      // on f-droid, so if txt is not found we look for yml as the fallback
+      result = await fetch.call(this, url, fetchOpts)
+    } catch (error) {
+      if (format) {
+        // if the format was specified it doesn't make the fallback request
+        throw error
+      }
+      result = await this.fetchYaml(url, fetchOpts)
+    }
+
+    return constructor.render(result)
+  }
+
+  async fetchYaml(url, options) {
+    const { buffer } = await this._request({
+      url: `${url}.yml`,
+      ...options,
+    })
+
+    // we assume the yaml structure as provided here:
+    // https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/org.dystopia.email.yml
+    try {
+      const { CurrentVersion: version } = yaml.safeLoad(
+        buffer.toString(),
+        'utf8'
+      )
+      if (!version) {
+        throw new Error('could not find version on website')
+      }
+      return { version }
+    } catch (error) {
+      throw new InvalidResponse({
+        prettyMessage: 'invalid response',
+        underlyingError: error,
+      })
+    }
+  }
+
+  async fetchText(url, options) {
+    const { buffer } = await this._request({
+      url: `${url}.txt`,
+      ...options,
     })
     const metadata = buffer.toString()
-    // we assume the layout as provided here:
+    // we assume the txt structure as provided here:
     // https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/axp.tool.apkextractor.txt
```

Comment:
Can the changes below now be reset?

Reply:
done https://github.com/badges/shields/pull/2377/commits/ed53088f3a3e31341fafac5051b2514cc832a6e4

Reply:
Sorry, I might not have been clear. I wasn't referring to the comments above, but to the changes in the code below which extract the version from the .txt metadata file. I think those changes were meant to accommodate the .yml format and are no longer needed. Is that right?

Reply:
yes, I was restore before to see this comment :D https://github.com/badges/shields/pull/2377/commits/3a745b694fd06c7e7fbba77fae9f80e0dff49e81